components/script/dom/uievent.rs

```diff
@@ -75,15 +75,32 @@ impl UIEvent {
         detail: i32,
         can_gc: CanGc,
     ) -> DomRoot<UIEvent> {
-        let ev = UIEvent::new_uninitialized_with_proto(window, proto, can_gc);
-        ev.InitUIEvent(
-            type_,
-            bool::from(can_bubble),
-            bool::from(cancelable),
-            view,
-            detail,
-        );
-        ev
+        UIEvent::initialize_ui_event(
+            window, proto, type_, can_bubble, cancelable, view, detail, can_gc,
+        )
+    }
+
+    /// based on <https://w3c.github.io/uievents/#initialize-a-uievent>
```

Comment:
```suggestion
    /// <https://w3c.github.io/uievents/#initialize-a-uievent>
```

Reply:
it's because this `initialize_ui_event` is not a plain impl of the spec, as described [here](https://github.com/servo/servo/pull/34524#discussion_r1874600447), will update as shared there.

Reply:
https://github.com/servo/servo/pull/34524/commits/41e209df4c3cbd7ed067c2fb071562f7903c267e